com.unity.netcode.gameobjects/Tests/Runtime/NetworkBehaviourUpdaterTests.cs

```diff
@@ -42,97 +42,112 @@ public override IEnumerator Setup()
+                // gathering netvars to test on
+                for (int i = 0; i < nbSpawnedObjects; i++)
                 {
-                    serverNetVarsToUpdate.AddRange(((INetVarInfo)networkBehaviour).AllNetVars);
-                    nbBehaviours++;
+                    var spawnedObject = Object.Instantiate(prefabToSpawn);
+                    var networkSpawnedObject = spawnedObject.GetComponent<NetworkObject>();
+                    networkSpawnedObject.NetworkManagerOwner = m_ServerNetworkManager;
+                    networkSpawnedObject.Spawn();
+                    int nbBehaviours = 0;
+                    foreach (var networkBehaviour in spawnedObject.GetComponents<NetworkBehaviour>())
+                    {
+                        serverNetVarsToUpdate.AddRange(((INetVarInfo)networkBehaviour).AllNetVars);
+                        nbBehaviours++;
+                    }
+                    Assert.That(nbBehaviours, Is.EqualTo((firstNetworkBehaviour == null ? 0 : 1) + (secondNetworkBehaviour == null ? 0 : 1)), "Setup: Did not find expected number of NetworkBehaviours");
                 }
-                Assert.That(nbBehaviours, Is.EqualTo((firstNetworkBehaviour == null ? 0 : 1) + (secondNetworkBehaviour == null ? 0 : 1)));
-            }
-            var serverNetVarCount = serverNetVarsToUpdate.Count;
 
-            yield return new WaitForSeconds(0); // wait a frame to make sure spawn is done
-            // todo: with Snapshot spawns enabled and the current race condition, the following line is needed:
-            // yield return new WaitForSeconds(0.2f); // wait a bit to fix the spawn/update race condition
+                yield return new WaitForSeconds(0); // wait a frame to make sure spawn is done
+                // todo: with Snapshot spawns enabled and the current race condition, the following line is needed:
+                // yield return new WaitForSeconds(0.2f); // wait a bit to fix the spawn/update race condition
 
-            foreach (var netVar in serverNetVarsToUpdate)
+                foreach (var netVar in serverNetVarsToUpdate)
+                {
+                    Assert.That(netVar.Value, Is.EqualTo(0), "Setup: Variables should start with 0 value"); // sanity check
+                }
+
+                setupSucceeded = true;
+            }
+            finally
             {
-                Assert.That(netVar.Value, Is.EqualTo(0)); // sanity check
+                Debug.Log(setupSucceeded ? "Test Setup Completed." : "Test Setup Failed.");
             }
 
             // test updating all netvars
-            int updatedValue = 1;
+            const int updatedValue = 1;
             foreach (var netVar in serverNetVarsToUpdate)
             {
                 netVar.Value = updatedValue;
-                Assert.That(netVar.IsDirty, Is.True);
+                // TODO: Why isn't this just a unit test?
```

Comment:
At the time, the goal was to test NetworkBehaviourUpdater, not to create unit tests for netvars if I remember correctly. But true, this better fits in individual netvar unit tests, it'd make knowing exactly what broke easier.